examples/connext_dds/waitset_status_cond/py/waitset_status_cond_publisher.py

```diff
@@ -21,25 +21,56 @@ def publisher_main(domain_id, sample_count):
     participant = dds.DomainParticipant(domain_id)
 
     # Create a Topic and automatically register the type
-    topic = dds.Topic(participant, "Example waitset_status_cond", Foo)
+    topic = dds.Topic(participant, "Example Foo", Foo)
 
-    # Create a DataWriter with default QoS (Publisher created in-line)
-    writer = dds.DataWriter(dds.Publisher(participant), topic)
+    # Create a DataWriter with default QoS
+    writer = dds.DataWriter(topic)
+
+    # Create a Status Condition for the writer
+    status_condition = dds.StatusCondition(writer)
+
+    # Enable statuses configuration for the Status Condition
+    status_condition.enabled_statuses = dds.StatusMask.PUBLICATION_MATCHED
+
+    # This enables start sending samples when we know there's a match
+    send_samples = False
+
+    # Define a handler for the Status Condition
+    def status_handler(_):
+        nonlocal send_samples
+        status_mask = writer.status_changes
+        st = writer.publication_matched_status
+
+        if dds.StatusMask.PUBLICATION_MATCHED in status_mask:
+            if st.current_count > 0:
```

Comment:
Print a message as you do in the subscriber

Reply:
And I wouldn't control the publication of data based on matched subscribers. That normally goes against the publish/subscribe design pattern.

Reply:
I was going for some "utility", but I'll simply print the information and continue publishing the data as usual